editor/src/clj/editor/app_view.clj

```diff
@@ -1870,6 +1870,20 @@ If you do not specifically require different script states, consider changing th
 (defn- tab->view-type [^Tab tab]
   (some-> tab (ui/user-data ::view-type) :id))
 
+(defn- handle-tab-pane-mouse-pressed!
+  "Selects the tab on right-click."
+  [^TabPane tab-pane ^MouseEvent event]
+  (when (= MouseButton/SECONDARY (.getButton event))
+    ;; Hackish way to get the tab index under the mouse cursor.
+    (when-let [target-tab (->> (.lookupAll tab-pane ".tab")
```

Comment:
Using `(.getTabs tab-pane)` instead of a lookup didn't work, because once again, tabs are not nodes.